submit_cgap/tests/test_submission.py

```diff
@@ -2202,3 +2253,144 @@ def test_running_on_windows_native():
+                file_name = "foo/bar"
+                input_arg = "foo"
+                function_wrapper = UploadMessageWrapper(uuid, no_query=no_query)
+                result = function_wrapper.wrap_upload_function(
+                    simple_function, file_name
+                )(input_arg, error_raised=error_raised)
+
+                expected_lines = []
+                if not no_query and cgap_selective_uploads and not yes_or_no_result:
+                    mocked_yes_or_no.assert_called_once()
+                    expected_lines.append("OK, not uploading it.")
+                    simple_function.assert_not_called()
+                else:
+                    expected_lines.append(f"Uploading {file_name} to item {uuid} ...")
+                    simple_function.assert_called_once_with(
+                        input_arg, error_raised=error_raised
+                    )
+                    if error_raised:
+                        expected_lines.append(f"RuntimeError: Error occurred")
+                    else:
+                        expected_lines.append(
+                            f"Upload of {file_name} to item {uuid} was successful."
+                        )
+                assert shown.lines == expected_lines
+                assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "credentials,files_found,expected_file_search_calls,expected_uploader_calls",
+    [
+        ([], [], 0, 0),
+        ([{"filename": "foo"}], [], 0, 0),
+        ([{"upload_credentials": {"key": "value"}}], [], 0, 0),
+        (SOME_EXTRA_FILE_CREDENTIALS, [], 2, 0),
+        (SOME_EXTRA_FILE_CREDENTIALS, [SOME_FILENAME], 2, 1),
+        (SOME_EXTRA_FILE_CREDENTIALS, [SOME_FILENAME, ANOTHER_FILE_NAME], 2, 2),
+    ]
+)
+def test_upload_extra_files(
+    credentials, files_found, expected_file_search_calls, expected_uploader_calls
+):
+    """Test extra files credentials utilized to search for and then
+    upload files.
+    """
+    folder = SOME_USER_HOMEDIR
+    recursive = True
+    auth = SOME_AUTH
+
+    def mocked_file_search(folder, extra_file_name, **kwargs):
+        if extra_file_name in files_found:
+            return os.path.join(folder, extra_file_name), None
+        else:
+            return None, "error"
+
+    mocked_search_for_file = mock.MagicMock(side_effect=mocked_file_search)
+    with mock.patch.object(
+        submission_module, "search_for_file", mocked_search_for_file
+    ):
+        with mock.patch.object(
+            submission_module, "execute_prearranged_upload"
+        ) as mocked_execute_upload:
+            uploader_wrapper = UploadMessageWrapper(SOME_UUID)
+            upload_extra_files(
+                credentials,
+                uploader_wrapper,
+                folder,
+                auth,
+                recursive=recursive,
+            )
+            assert len(
+                mocked_search_for_file.call_args_list
+            ) == expected_file_search_calls
```

Comment:
Are you using a scrunchy editor? We usually assume 120 character lines and I _think_ this fits fine on one line.
```suggestion
            assert len(mocked_search_for_file.call_args_list) == expected_file_search_calls
```